engine-nextgen/src/main/kotlin/graphql/nadel/NextgenEngine.kt

```diff
@@ -74,14 +78,16 @@ class NextgenEngine @JvmOverloads constructor(nadel: Nadel, transforms: List<Nad
             executionInput.variables,
         )
 
+        val executionContext = NadelExecutionContext(executionInput, serviceExecutionHooks)
```

Comment:
Moves the instantiation of the `NadelExecutionContext` up, to ensure we have one instance per execution